generator/src/lib.rs

```diff
@@ -2133,6 +2134,32 @@ pub fn write_source_code(path: &Path) {
         use std::fmt;
         use std::os::raw::*;
 
+        #[repr(C)]
+        #[derive(Debug, Copy, Clone)]
+        pub struct ExtensionChain {
+            pub s_type: StructureType,
+            pub p_next: *mut c_void
+        }
```

Comment:
Shouldn't this be named `BaseStructure` or similar, along the lines of upstream?

Reply:
That makes sense; then the loop helper can just be a top-level function, which perhaps shouldn't even be exported; it doesn't seem very broadly useful, and reducing API surface area is nice.